compiler/rustc_mir/src/interpret/cast.rs

```diff
@@ -142,6 +144,17 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // Handle cast from a univariant (ZST) enum.
         match src.layout.variants {
             Variants::Single { index } => {
+                // This is dead code, because an uninhabited enum is UB to
+                // instantiate. But we want to behave nicely, so just return a 0
+                // value for the discriminant.
+                if let ty::Adt(adt, _) = src.layout.ty.kind {
```

Comment:
Usually we prefer to test the layout instead of the type, so can you check `src.layout.abi.is_uninhabited()` instead?